engine/src/main/java/org/hibernate/validator/internal/util/ReflectionHelper.java

```diff
@@ -259,36 +260,27 @@ public static Object getValue(Method method, Object object) {
 	}
 
 	/**
-	 * Indicates if the type is considered indexable (ie is an {@code Iterable}, an array or a {@code Map}).
+	 * Indicates if the type is considered a collection of elements (ie is an <code>Iterable</code>, an array or
+	 * a <code>Map</code>).
 	 *
-	 * @param type the type to inspect.
-	 *
-	 * @return Returns true if the type is indexable.
+	 * @param type the type to check.
+	 * @return returns <code>true</code> if <code>type</code> can be considered a collection of elements
```

Comment:
I'd prefer more useful docs over making a tool happy. I don't know whether it's possible, but maybe one could have different rules/policies for API/SPI (where I can get behind a 0 warnings policy) and internal parts (where that seems over the top).